src/Microsoft.Android.Build.BaseTasks/AndroidToolTask.cs

```diff
@@ -12,21 +14,35 @@ public abstract class AndroidToolTask : ToolTask
 
 		protected string WorkingDirectory { get; private set; }
 
+		StringBuilder toolOutput;
+
 		public AndroidToolTask ()
 		{
 			WorkingDirectory = Directory.GetCurrentDirectory();
+			toolOutput = new StringBuilder ();
 		}
 
 		public override bool Execute ()
 		{
 			try {
-				return RunTask ();
+				bool taskResult = RunTask ();
+				if (!taskResult && !string.IsNullOrEmpty (toolOutput?.ToString ())) {
+					Log.LogError (toolOutput.ToString ().Trim ());
```

Comment:
I feel like this one should have an error code, can we use `$"XA{TaskPrefix}0000"`, maybe we can add new helper method in this class?

https://github.com/xamarin/xamarin-android-tools/blob/37d79c9dcdf738a181084b0b5890877128d75f1e/src/Microsoft.Android.Build.BaseTasks/UnhandledExceptionLogger.cs#L24

Reply:
Thanks, I've attempted to address this with the latest changes to use LogCodedError instead

Reply:
I was reading what @jonpryor  wrote on #208:

> Update AndroidToolTask so that it provides output logging infrastructure, a'la https://github.com/xamarin/xamarin-android/commit/b002dc38, such that all tool output is always captured (as well as logged via LogDebugMessage()), and when an error occurs, all captured output is part of the error message.

Do we somehow need to "smush" the extra error information into an existing error message? 🤦 This PR would two errors, right?

Reply:
The current approach should indeed show two errors, though I think both are useful. Here is a sample of complete build output from the test case that was added:

```
    "OutputTest.proj" (Test target) (1:2) ->
    (Test target) -> 
      OutputTest.proj(10,9): error MSB6006: "dotnet" exited with code 1.
      OutputTest.proj(10,9): error XADTOT0000: Could not execute because the specified command or file was not found.
    OutputTest.proj(10,9): error XADTOT0000: Possible reasons for this include:
    OutputTest.proj(10,9): error XADTOT0000:   * You misspelled a built-in dotnet command.
    OutputTest.proj(10,9): error XADTOT0000:   * You intended to execute a .NET program, but dotnet-invalidcommand does not exist.
    OutputTest.proj(10,9): error XADTOT0000:   * You intended to run a global tool, but a dotnet-prefixed executable with this name could not be found on the PATH.

0 Warning(s)
2 Error(s)
```

The first error shows the tool name and exit code, and the second error is our new custom error with all of the output from the tool. Do we want to try to suppress/replace the MSB6006 error?